File: OpenEdXMobile/src/main/java/org/edx/mobile/http/interceptor/OauthHeaderRequestInterceptor.java
package org.edx.mobile.http.interceptor;

import android.content.Context;
import androidx.annotation.NonNull;

import org.edx.mobile.logger.Logger;
import org.edx.mobile.module.prefs.LoginPrefs;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import roboguice.RoboGuice;

/**
 * Injects OAuth token - if present - into Authorization header
 **/
public final class OauthHeaderRequestInterceptor implements Interceptor {
    protected final Logger logger = new Logger(getClass().getName());

    @NonNull
    private final LoginPrefs loginPrefs;

    public OauthHeaderRequestInterceptor(@NonNull Context context) {
        loginPrefs = RoboGuice.getInjector(context).getInstance(LoginPrefs.class);
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        final Request.Builder builder = chain.request().newBuilder();
        final String token = loginPrefs.getAuthorizationHeader();
        if (token != null) {
            builder.addHeader("Authorization", token);
        }
        return chain.proceed(builder.build());
    }

}

File: OpenEdXMobile/src/main/java/org/edx/mobile/http/interceptor/OauthHeaderRequestInterceptor.kt
package org.edx.mobile.http.interceptor

import android.content.Context
import dagger.hilt.android.EntryPointAccessors
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import org.edx.mobile.core.EdxDefaultModule.ProviderEntryPoint
import org.edx.mobile.module.prefs.LoginPrefs
import org.edx.mobile.util.AppConstants.HEADER_KEY_AUTHORIZATION
import java.io.IOException

/**
 * Injects OAuth token into Authorization header if present
 */
class OauthHeaderRequestInterceptor(
    context: Context
) : Interceptor {
    private val loginPrefs: LoginPrefs

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder: Request.Builder = chain.request().newBuilder()
        val token = loginPrefs.authorizationHeader
        if (token != null) {
            builder.addHeader(HEADER_KEY_AUTHORIZATION, token)
        }
        return chain.proceed(builder.build())
    }

    init {
        loginPrefs = EntryPointAccessors.fromApplication(
            context,
            ProviderEntryPoint::class.java
        ).getLoginPrefs()
    }
}
